app/src/main/kotlin/com/wire/android/ui/userprofile/image/AvatarPicker.kt

```diff
@@ -37,19 +37,22 @@ import com.wire.android.ui.common.textfield.WirePrimaryButton
 import com.wire.android.ui.common.topappbar.WireCenterAlignedTopAppBar
 import com.wire.android.ui.theme.wireColorScheme
 import com.wire.android.ui.userprofile.image.AvatarPickerViewModel.ErrorCodes
-import com.wire.android.util.getWritableTempAvatarUri
+import com.wire.android.util.AvatarImageManager.Companion.getShareableTempAvatarUri
 
 @OptIn(ExperimentalMaterialApi::class, ExperimentalMaterial3Api::class)
 @Composable
 fun AvatarPickerScreen(viewModel: AvatarPickerViewModel) {
     val context = LocalContext.current
-    val state = rememberAvatarPickerState()
-
-    // We need to launch an effect to update the initial avatar uri whenever the pickerVM updates successfully the raw image
-    LaunchedEffect(viewModel.avatarRaw) {
-        val currentAvatarUri = getWritableTempAvatarUri(viewModel.avatarRaw ?: ByteArray(16), context)
-        state.avatarPickerFlow.pictureState = PictureState.Initial(currentAvatarUri)
-    }
+    val targetAvatarUri = getShareableTempAvatarUri(context)
+    val state = rememberAvatarPickerState(
+        onImageSelected = { viewModel.pickNewImage(it) },
+        onPictureTaken = { wasSaved ->
+            if (wasSaved) {
+                viewModel.postProcessAvatarImage(targetAvatarUri)
+            }
+        },
+        targetPictureFileUri = targetAvatarUri
```

Comment:
are we going to set this targetAvatarUri from this composable, like it could come from the viewmodel ? if not, I think we could call this method  getShareableTempAvatarUri inside rememberAvatarPickerState()